persistence/rocksdb/src/main/java/org/infinispan/persistence/rocksdb/RocksDBStore.java

```diff
@@ -195,7 +196,7 @@ private void migrateFromV11() throws IOException, RocksDBException {
          WriteBatch batch = new WriteBatch();
          Set<MarshallableEntry<K, V>> expirableEntries = new HashSet<>();
          Flowable.fromPublisher(publisher)
-               .subscribe(e -> {
+               .blockingSubscribe(e -> {
```

Comment:
This is my lack of knowledge. Does the blocking subscribe happen on the same threads from the `publisher`? Should we set which scheduler to observe or does it utilize the blocking manager already?

Reply:
The callback is invoked in the current thread and the method waits for everything to complete. This method is invoked by a blocking thread anyway:
https://github.com/infinispan/infinispan/blob/e9d227d6df1e9a1a5034737dbab2c37b2c82a7d1/persistence/rocksdb/src/main/java/org/infinispan/persistence/rocksdb/RocksDBStore.java#L143

Reply:
A publisher by default will subscribe on the thread that calls subscribe. Looking at the code, it appears it was relying upon this before, but the publisher method passes off the calls to the blocking manager, so that was not guaranteed. Looking at this it looks like it was very easy to cause a crash or not to migrate data properly, so I am surprised this hasn't come up before even.

Looking at this method though this is a really scary migration as EVERY entry in the store will be temporarily stored in memory, this should have really done some sort of batched update instead. But given we don't really recommend rocks db anymore, I guess it is fine.